frontend/src/Frontend/UI/Wallet.hs

```diff
@@ -70,14 +71,25 @@ type HasUiWalletModelCfg mConf key m t =
 
 -- | UI for managing the keys wallet.
 uiWallet
-  :: (MonadWidget t m, HasUiWalletModelCfg mConf key m t)
+  :: forall m t key mConf
+     . ( MonadWidget t m
+       , HasUiWalletModelCfg mConf key m t
+       )
   => Wallet key t
-  -> m mConf
-uiWallet w = divClass "keys group" $ do
-    onCreate <- uiCreateKey w
-    keysCfg <- uiAvailableKeys w
-
-    pure $ keysCfg & walletCfg_genKey .~ fmap (\a -> (a, "0", "")) onCreate -- TODO let user pick chain/notes
+  -> ModalIde m key t
+  -> m (mConf, ModalIdeCfg m key t)
+uiWallet w m = divClass "keys group" $ do
+  eOpenAddAccount <- confirmButton def "+ Add Account"
+
+  onCreate <- uiCreateKey w
+  keysCfg <- uiAvailableKeys w
+
+  pure ( keysCfg & walletCfg_genKey .~ fmap (\a -> (a, "0", "")) onCreate -- TODO let user pick chain/notes
+       , mempty & modalCfg_setModal .~ leftmost
+         [ Just (uiCreateWalletOnlyAccount m) <$ eOpenAddAccount
+         , Just . uiWalletOnlyAccountCreated <$> (w ^. wallet_walletOnlyAccountCreated)
```

Comment:
The presence of two dialogs here seems off, I think we should just have one which contains a workflow.